cypress/e2e/shared/scriptQueryBuilder.results.test.ts

```diff
@@ -0,0 +1,366 @@
+      })
+    })
+
+    it('will allow querying of different data ranges', () => {
+      selectSchema()
+      confirmSchemaComposition()
+
+      cy.log('query with default date range of -1h')
+      cy.getByTestID('time-machine-submit-button').should('exist').click()
+      cy.wait('@query -1h')
+
+      cy.log('query date range can be adjusted')
+      cy.getByTestID('timerange-dropdown').within(() => {
+        cy.getByTestID('dropdown--button').should('exist')
+        cy.getByTestID('dropdown--button').clickAttached()
+      })
+      cy.getByTestID('dropdown-item-past15m').should('exist').click()
+      cy.getByTestID('time-machine-submit-button').should('exist').click()
+      cy.wait('@query -15m')
+    })
+
+    describe('data completeness', () => {
+      const downloadsDirectory = Cypress.config('downloadsFolder')
+      const browser = Cypress.config('browser')
+
+      const validateCsv = (csv: string, tableCnt: number) => {
+        cy.wrap(csv)
+          .then(doc => doc.trim().split('\n'))
+          .then(list => {
+            expect(list[list.length - 1]).contains(`,${tableCnt - 1},`)
+          })
+      }
+
+      const runTest = (
+        tableCnt: number,
+        rowCnt: number,
+        truncated: boolean
+      ) => {
+        cy.log('confirm on 1hr')
+        cy.getByTestID('timerange-dropdown').within(() => {
+          cy.getByTestID('dropdown--button').should('exist')
+          cy.getByTestID('dropdown--button').clickAttached()
+        })
+        cy.getByTestID('dropdown-item-past1h').should('exist')
+        cy.getByTestID('dropdown-item-past1h').clickAttached()
+        cy.getByTestID('time-machine-submit-button').should('exist')
+        cy.getByTestID('time-machine-submit-button').clickAttached()
+
+        cy.log('run query')
+        cy.getByTestID('time-machine-submit-button').should('exist')
+        cy.getByTestID('time-machine-submit-button').clickAttached()
+        cy.wait('@query -1h')
+        cy.wait(1000) // bit more time for csv parsing
+
+        cy.log('table metadata displayed to user is correct')
+        if (truncated) {
+          cy.getByTestID('query-stat')
+            .should('contain', 'truncated')
+            .should('not.contain', 'rows')
+            .should('not.contain', 'tables')
+        } else {
+          cy.getByTestID('query-stat')
+            .should('not.contain', 'truncated')
+            .contains(`${tableCnt} tables`)
+          cy.getByTestID('query-stat').contains(`${rowCnt} rows`)
+        }
+
+        cy.log('will download complete csv data')
+        cy.intercept('POST', '/api/v2/query?*', req => {
+          req.redirect(route)
+        }).as('queryDownloadCSV')
+
+        // TODO: debug intermittent failures which occur 30% of the time
+        // in firefox only, in the CI only, and only when no data is returned
+        if (browser.name.toLowerCase() != 'chrome') {
+          return
+        }
```

Comment:
Do we still need this? We turned off firefox test runs, right?

Reply:
Good catch. Is copy/pasta. Will remove.